Lib/functools.py

```diff
@@ -1042,3 +1037,24 @@ def __get__(self, instance, owner=None):
         return val
 
     __class_getitem__ = classmethod(GenericAlias)
+
+def _warn_kwargs(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        if kwargs:
+            import os
+            import warnings
+            warnings.warn("""\
+Calling functools.reduce with keyword arguments is deprecated
+in Python 3.14 and will be forbidden in Python 3.16.""",
+                         DeprecationWarning,
+                         skip_file_prefixes=(os.path.dirname(__file__),))
```

Comment:
Why skipping the warning when the function is called in the stdlib?

Reply:
Without skipping output will be less prettier.
Without `skip_file_prefixes`:
```python
>>> functools.reduce(function=lambda x, y: 1, sequence=[1, 2])
/Users/admin/Projects/cpython/Lib/functools.py:1047: DeprecationWarning: Calling functools.reduce with keyword arguments is deprecated in Python 3.14 and will be forbidden in Python 3.16.
  warnings.warn(
1
```

With `skip_file_prefixes`:
```python
>>> functools.reduce(function=lambda x, y: 1, sequence=[1, 2])
<python-input-1>:1: DeprecationWarning: Calling functools.reduce with keyword arguments is deprecated in Python 3.14 and will be forbidden in Python 3.16.
  functools.reduce(function=lambda x, y: 1, sequence=[1, 2])
1
```

Reply:
Ah, only the filename is skipped, not the warning, ok.